06_miscellaneous/versioning_slides.md

```diff
@@ -0,0 +1,137 @@
+---
+type: slide
+slideOptions:
+  transition: slide
+  width: 1400
+  height: 900
+  margin: 0.1
+---
+
+<style>
+  .reveal strong {
+    font-weight: bold;
+    color: orange;
+  }
+  .reveal p {
+    text-align: left;
+  }
+  .reveal section h1 {
+    color: orange;
+  }
+  .reveal section h2 {
+    color: orange;
+  }
+  .reveal code {
+    font-family: 'Ubuntu Mono';
+    color: orange;
+  }
+  .reveal section img {
+    background:none;
+    border:none;
+    box-shadow:none;
+  }
+</style>
+
+# Versioning
+
+---
+
+## Introduction
+
+- Software version may indicate
+    - Currentness/"Freshness"
+    - (In)compatibility
+    - Significance of changes
+- Easy-to-understand label that something changed
+- Important for packaging and reproducibility
+
+---
+
+## Versioning Examples
+
+- Semantic versioning
+    - [Linux Kernel](https://kernel.org/): 5.16.5 (Release), 5.17-rc2 (Preview)
+- [PEP 440](https://www.python.org/dev/peps/pep-0440/)
+    - [pyMOR](https://pymor.org/): 2021.2.1
+- Calendar versioning
+    - [Ubuntu](ubuntu.com/): 21.10 (Latest), 20.04 (LTS)
+    - [OpenFOAM](https://www.openfoam.com/current-release) (`.com`): v2112
+- Others
+    - [TeX](https://tug.org/): 3.141592653 (Latest), 3.14159265 (Ubuntu 20.04)
+    - [OpenFOAM](https://openfoam.org/release/) (`.org`): 9 (Latest), but "always releasable"
+
+---
+
+## Versioning standards
+
+- [Semantic Versioning](https://semver.org/)
+- [Calendar Versioning](https://calver.org/)
+- [PEP 440](https://www.python.org/dev/peps/pep-0440/)
+
+---
+
+## Semantic Versioning
+
+```text
+MAJOR.MINOR.PATCH-<pre-release>+<build>
+```
+
+- `MAJOR` version when you make incompatible API changes,
```

Comment:
No need to write this down necessarily, but I guess we might need to explain here a bit:

a) what incompatible changes are (could also be more than API, e.g. configuration files)
b) why they are bad -> everything that depends on you needs to update